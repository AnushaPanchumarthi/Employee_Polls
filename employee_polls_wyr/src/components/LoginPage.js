import React, { useState } from 'react';
import { connect } from 'react-redux';
import { useRef } from 'react';
import { setAuthedUser } from '../actions/authedUser';
import Illustration from '../images/WYR.jpg';
import {  useLocation, useNavigate } from 'react-router-dom';

const LoginPage = (props) => {
  const navigate = useNavigate();
  const { state } = useLocation();
  const dispatch = props.dispatch;
  const userIds = props.users;
  const inputRef = useRef();
  const [selectedUser, setSelectedUser] = useState(userIds[0]);

  const changeUser = (e) => {
    setSelectedUser(inputRef.current.value);
  };
  const handleLogin = (e) => {
    e.preventDefault();
    dispatch(setAuthedUser(inputRef.current.value));
    if (inputRef.current.value || selectedUser) {
      return navigate(state?.path || "/",{'replace': false});
    }
  };
  return (
    <section className="py-sm-5">
      <div className="container px-3">
        <div className="row align-items-center" style={{ textAlign: 'center' }}>
          <div className="col-md-6 mt-2 text-center">
            <img
              align="left"
              src={Illustration}
              margin="10%"
              width="90%"
              alt="Poll Now"
            />
          </div>
          <div className="col-md-6 mt-4" margin="100%">
            <h1 className="login" style={{ fontSize: '30px' }}>
              Login! for Employee Polls
            </h1>
            <h2 style={{ color: 'red', fontSize: '20px' }}>
              {' '}
              Would you Rather!!{' '}
            </h2>
            <form onSubmit={handleLogin}>
              <div className="field">
                <label style={{ display: 'block', fontSize: '20px' }}>
                  {' '}
                  Select The User from the list{' '}
                </label>
                <select
                  style={{
                    padding: '.5rem .8rem .5rem .8rem',
                    margin: '20px 5px'
                  }}
                  size="lg"
                  id="login"
                  name="Username"
                  data-testid="username-select"
                  onChange={changeUser}
                  ref={inputRef}
                  required
                >
                  <option key="initalText" disabled>
                    ChooseUser
                  </option>
                  {userIds.map((userId) => (
                    <option key={userId} value={userId}>
                      {userId}
                    </option>
                  ))}
                </select>
              </div>
              <button
                className="btn  btn-dark px-4 my-2"
                type="submit"
                data-testid="submit-button" >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </section>
  );
};

const mapStateToProps = ({ users }) => {
  return {
    users: Object.keys(users)
  };
};

export default connect(mapStateToProps)(LoginPage);
