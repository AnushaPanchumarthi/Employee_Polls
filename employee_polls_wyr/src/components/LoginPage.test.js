import { render, fireEvent, screen } from '@testing-library/react';
import LoginPage from './LoginPage';
import { MemoryRouter as Router } from 'react-router-dom';
import reducer from '../reducers';
import middleware from '../middleware';
import { createStore } from 'redux';
import { Provider } from 'react-redux';

describe('Login test scenarios', () => {
  /*snapshot testing*/
  it('matches the snapshot when a user logs in', () => {
    const store = createStore(reducer, middleware);
    const component = render(
      <Provider store={store}>
        <Router>
          <LoginPage />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  it('Succesfull Login', () => {
    const store = createStore(reducer, middleware);
    const component = render(
      <Provider store={store}>
        <Router>
          <LoginPage />
        </Router>
      </Provider>
    );
    var username = component.queryByTestId('username-select');
    fireEvent.change(username, { target: { value: 'mtsamis' } });
    var submitButton = component.getByTestId('submit-button');
    fireEvent.click(submitButton);
    screen.debug();
  });
});
