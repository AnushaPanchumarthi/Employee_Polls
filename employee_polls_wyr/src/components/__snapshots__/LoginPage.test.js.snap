// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login test scenarios matches the snapshot when a user logs in 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="py-sm-5"
      >
        <div
          class="container px-3"
        >
          <div
            class="row align-items-center"
            style="text-align: center;"
          >
            <div
              class="col-md-6 mt-2 text-center"
            >
              <img
                align="left"
                alt="Poll Now"
                margin="10%"
                src="WYR.jpg"
                width="90%"
              />
            </div>
            <div
              class="col-md-6 mt-4"
              margin="100%"
            >
              <h1
                class="login"
                style="font-size: 30px;"
              >
                Login! for Employee Polls
              </h1>
              <h2
                style="color: red; font-size: 20px;"
              >
                 Would you Rather!! 
              </h2>
              <form>
                <div
                  class="field"
                >
                  <label
                    style="display: block; font-size: 20px;"
                  >
                     Select The User from the list 
                  </label>
                  <select
                    data-testid="username-select"
                    id="login"
                    name="Username"
                    required=""
                    style="padding: .5rem .8rem .5rem .8rem; margin: 20px 5px;"
                  >
                    <option
                      disabled=""
                    >
                      ChooseUser
                    </option>
                  </select>
                </div>
                <button
                  class="btn  btn-dark px-4 my-2"
                  data-testid="submit-button"
                  disabled=""
                  type="submit"
                >
                  Login
                </button>
              </form>
            </div>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="py-sm-5"
    >
      <div
        class="container px-3"
      >
        <div
          class="row align-items-center"
          style="text-align: center;"
        >
          <div
            class="col-md-6 mt-2 text-center"
          >
            <img
              align="left"
              alt="Poll Now"
              margin="10%"
              src="WYR.jpg"
              width="90%"
            />
          </div>
          <div
            class="col-md-6 mt-4"
            margin="100%"
          >
            <h1
              class="login"
              style="font-size: 30px;"
            >
              Login! for Employee Polls
            </h1>
            <h2
              style="color: red; font-size: 20px;"
            >
               Would you Rather!! 
            </h2>
            <form>
              <div
                class="field"
              >
                <label
                  style="display: block; font-size: 20px;"
                >
                   Select The User from the list 
                </label>
                <select
                  data-testid="username-select"
                  id="login"
                  name="Username"
                  required=""
                  style="padding: .5rem .8rem .5rem .8rem; margin: 20px 5px;"
                >
                  <option
                    disabled=""
                  >
                    ChooseUser
                  </option>
                </select>
              </div>
              <button
                class="btn  btn-dark px-4 my-2"
                data-testid="submit-button"
                disabled=""
                type="submit"
              >
                Login
              </button>
            </form>
          </div>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
